main.cpp

```diff
@@ -414,6 +549,28 @@ int Oscillator::makeOscillatorSquareWave(int oscillatorWave)
     return oscillatorWave + squareWave;
 }
 
+std::vector<int> Oscillator::populateWaveWithData()
+{
+    int container[10] = {0, 3, 7, 7, 10, 4, 5, 6, 3, 8};
```

Comment:
don't use raw arrays.
This isn't C.
this is C++
use containers like std::array or std::vector<> and give yourself some bounds checking safety.
- [ ] revise

Reply:
You need to replace this raw array with something else
- [ ] revise